File: notification_service/python/models/notificationModel.py
from pymongo import MongoClient
from bson.objectid import ObjectId
from datetime import datetime

client = MongoClient('mongodb://localhost:27017/')
db = client['notification_database']
notifications = db['notifications']

def create_notification(user_id, message):
    notification = {
        "notification_id": ObjectId(),
        "user_id": ObjectId(user_id),
        "message": message,
        "read": False,
        "created_at": datetime.utcnow()
    }
    return notifications.insert_one(notification)

def mark_as_read(notification_id):
    return notifications.update_one({"notification_id": ObjectId(notification_id)}, {"$set": {"read": True}})

File: services/notification_service/models/notificationModel.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const notificationSchema = new Schema({
    notification_id: { type: Schema.Types.ObjectId, required: true },
    user_id: { type: Schema.Types.ObjectId, required: true },
    message: { type: String, required: true },
    read: { type: Boolean, default: false },
    created_at: { type: Date, default: Date.now }
});

const Notification = mongoose.model('Notification', notificationSchema);

module.exports = Notification;
